### PR TITLE
Granular deletion and visibility setting based on user preference

### PR DESCRIPTION
**Pull request by rtibbles**
### Summary
* Updates importcontent and annotation machinery to only mark visibility (`ContentNode.available`) in ContentNodes defined by `node_ids` and `exclude_node_ids`.
* Update deletechannel to allow a`node_ids` argument to delete specific nodes within a channel, and not the entire channel.
* Removes the `available` field on `File` as it was unnecessary.
* Renames some annotation functions to be more semantic - focus on '`ContentNode` visibility'
* Adds `node_ids` as a param to the deletion endpoints
* Cleans up and consolidates task validation
* Clears file_size and resource_count estimates for granular deletion - will need follow up work to update this on the task metadata from within the job once #6007 is merged.

### Reviewer guidance
* When you import a topic inside Khan Academy Arithmetic, does only the Arithmetic topic appear inside the Math topic?
* Can you granularly delete just one topic using the `deletecontent` management command now?

### References
https://www.notion.so/learningequality/Post-import-content-management-c1bba77a25ce438692ed6bc9b9f0d30c

----

### Contributor Checklist

PR process:

- [x] PR has the correct target branch and milestone
- [x] PR has 'needs review' or 'work-in-progress' label
- [x] If PR is ready for review, a reviewer has been added. (Don't use 'Assignees')
- [ ] If this is an important user-facing change, PR or related issue has a 'changelog' label
- [ ] If this includes an internal dependency change, a link to the diff is provided

Testing:

- [x] Contributor has fully tested the PR manually
- [ ] If there are any front-end changes, before/after screenshots are included
- [ ] Critical user journeys are covered by Gherkin stories
- [x] Critical and brittle code paths are covered by unit tests

### Reviewer Checklist

- Automated test coverage is satisfactory
- PR is fully functional
- PR has been tested for [accessibility regressions](http://kolibri-dev.readthedocs.io/en/develop/manual_testing.html#accessibility-a11y-testing)
- External dependency files were updated if necessary (`yarn` and `pip`)
- Documentation is updated
- Contributor is in AUTHORS.md

### CODE REVIEW

File: kolibri/core/content/base_models.py [hunk 1]
```diff
@@ -156,7 +156,6 @@ class File(models.Model):
     id = UUIDField(primary_key=True)
     # The foreign key mapping happens here as many File objects can map onto a single local file
     local_file = models.ForeignKey("LocalFile", related_name="files")
-    available = models.BooleanField(default=False)
```
Comment (indirectlylit) [inline kolibri/core/content/base_models.py:159]: nice

File: kolibri/core/content/management/commands/deletechannel.py [hunk 2]
```diff
@@ -14,9 +15,46 @@
 class Command(AsyncCommand):
     def add_arguments(self, parser):
         parser.add_argument("channel_id", type=str)
+        # However, some optional arguments apply to both groups. Add them here!
+        node_ids_help_text = """
+        Specify one or more node IDs to delete. Only these ContentNodes and descendants will be deleted.
+
+        e.g.
+
+        kolibri manage deletechannel --node_ids <id1>,<id2>, [<ids>,...] <channel id>
```
Comment (indirectlylit) [inline kolibri/core/content/management/commands/deletechannel.py]:
```suggestion
        kolibri manage deletechannel --node_ids <id1>,<id2>,[<ids>,...] <channel id>
```

File: kolibri/core/content/management/commands/deletechannel.py [hunk 3]
```diff
@@ -14,9 +15,46 @@
...
+            dest="node_ids",
+            help=node_ids_help_text,
+        )
+
+        exclude_node_ids_help_text = """
+        Specify one or more node IDs to exclude. Descendants of these node IDs will be not be deleted.
+
+        e.g.
+
+        kolibri manage deletechannel --exclude_node_ids <id1>,<id2>, [<ids>,...] <channel id>
```
Comment (indirectlylit) [inline kolibri/core/content/management/commands/deletechannel.py]:
```suggestion
        kolibri manage deletechannel --exclude_node_ids <id1>,<id2>,[<ids>,...] <channel id>
```

File: kolibri/core/content/management/commands/deletechannel.py [hunk 4]
```diff
@@ -14,9 +15,46 @@
 class Command(AsyncCommand):
     def add_arguments(self, parser):
         parser.add_argument("channel_id", type=str)
+        # However, some optional arguments apply to both groups. Add them here!
+        node_ids_help_text = """
+        Specify one or more node IDs to delete. Only these ContentNodes and descendants will be deleted.
```
Comment (indirectlylit) [inline kolibri/core/content/management/commands/deletechannel.py]: perhaps rename the command to `deletecontent`, and add an alias to `deletechannel` for backward compatibility?

**Review by indirectlylit — COMMENTED** [review]
cool

File: kolibri/core/content/management/commands/deletechannel.py (hunk 4 above)
Comment (rtibbles) [reply kolibri/core/content/management/commands/deletechannel.py]: Yeah, I had a similar thought. Will update.

Comment (indirectlylit) [reply kolibri/core/content/management/commands/deletechannel.py]: > I had a similar thought

Indeed you did :)

image

File: kolibri/core/content/management/commands/deletechannel.py [hunk 5]
```diff
@@ -14,9 +15,46 @@
...
+            default=[],
+            required=False,
+            dest="exclude_node_ids",
+            help=exclude_node_ids_help_text,
+        )
 
     def handle_async(self, *args, **options):
         channel_id = options["channel_id"]
+        node_ids = options["node_ids"]
+        exclude_node_ids = options["node_ids"]
```
Comment (jonboiser) [inline kolibri/core/content/management/commands/deletechannel.py]: `options["exclude_node_ids"]` here

Comment (jonboiser) [inline kolibri/core/content/management/commands/deletechannel.py]:
```suggestion
        exclude_node_ids = options["exclude_node_ids"]
```

Comment (rtibbles) [reply kolibri/core/content/management/commands/deletechannel.py]: Whoops - copypasta error.

**Review by indirectlylit — APPROVED** [review]
forward!

**Pull request merged**